lib/lab_manager/provider/vsphere.rb

```diff
@@ -105,7 +108,20 @@ def create_vm(opts = {})
     end
 
     def terminate_vm(_opts = {})
-      VSphere.connect.with do |_vs|
+      fail ArgumentError, 'Virtual machine data not present' unless instance_uuid
+
+      VSphere.connect.with do |vs|
+        Retryable.retryable(tries: 3) do
+          unless vs.get_virtual_machine(instance_uuid)['power_state'] == 'poweredOff'
+            vs.vm_power_off('force' => true, 'instance_uuid' => instance_uuid)
+          end
```

Comment:
I am not sure about this retry block. Isn't better: 
* use wait_for_completition
* check resutl of vm_power_off operation
Don't know if use 'force' => true, as I see https://github.com/fog/fog/blob/master/lib/fog/vsphere/models/compute/server.rb#L91-L95 they use force according to vmtools. And thats mean false if they are installed and running. ...don't know why.